ccm_web/client/src/components/SectionSelectorWidget.tsx

```diff
@@ -123,32 +138,23 @@ function SectionSelectorWidget (props: ISectionSelectorWidgetProps): JSX.Element
   // The debounced version of the text in the search field
   // Changes here will be passed along to sectionSearcherText
   const [searchFieldTextDebounced, setSearchFieldTextDebounced] = useDebounce<string | undefined>(undefined, 750)
-  const [internalSections, setInternalSections] = useState<SelectableCanvasCourseSection[]>(props.sections)
 
+  const [internalSections, setInternalSections] = useState<SelectableCanvasCourseSection[]>(props.sections)
   const [isSelectAllChecked, setIsSelectAllChecked] = useState<boolean>(false)
-
   const [anchorSortEl, setAnchorSortEl] = useState<null | HTMLElement>(null)
 
   const [searcher, setSearcher] = useState<ISectionSearcher | undefined>(props.search.length > 0 ? (props.search)[0] : undefined)
   const [searchFieldLabel, setSearchFieldLabel] = useState<string | undefined>(props.search.length > 0 ? (props.search)[0].helperText : undefined)
 
-  const [sectionsToUnmerge, setSectionsToUnmerge] = useState<CanvasCourseSectionWithCourseName[]>([])
   const [doUnmerge, isUnmerging, unmergeError] = usePromise(
-    async () => await unmergeSections(sectionsToUnmerge)
+    async (sections: CanvasCourseSectionWithCourseName[]) => await unmergeSections(sections),
+    (unmergedSections: CanvasCourseSectionBase[]) => {
+      const unmergedSectionIds = unmergedSections.map(s => s.id)
+      setInternalSections(internalSections.filter(section => !unmergedSectionIds.includes(section.id)))
+      if (props.sectionsRemoved !== undefined) props.sectionsRemoved(unmergedSections)
+    }
   )
 
-  useEffect(() => {
-    if (sectionsToUnmerge.length > 0) {
-      // eslint-disable-next-line @typescript-eslint/no-floating-promises
-      doUnmerge().then(() => {
-        setInternalSections(internalSections.filter(section => { return !sectionsToUnmerge.map(s2u => { return s2u.id }).includes(section.id) }))
-        if (props.sectionsRemoved !== undefined) {
-          props.sectionsRemoved(sectionsToUnmerge)
-        }
-        setSectionsToUnmerge([])
-      })
-    }
-  }, [sectionsToUnmerge])
```

Comment:
See 1 in PR description.

Reply:
Is this change needed as part of accessibility fix #260? or you felt this is simple/improved way of handing the click event so did it?

Reply:
Not part of the accessibility fix, just something that made sense to me.